presto-tests/src/test/java/com/facebook/presto/tests/TestTpchQueriesSelectivityStats.java

```diff
@@ -0,0 +1,194 @@
+                        " AND o_orderdate < DATE '1993-07-01' + INTERVAL '3' MONTH",
+                checks -> checks.estimate(OUTPUT_ROW_COUNT, defaultTolerance()));
+    }
+
+    @Test(enabled = false)
+    void testLineitemQ4()
+    {
+        statisticsAssertion.check(
+                "SELECT * FROM lineitem WHERE l_commitdate < l_receiptdate",
+                checks -> checks.estimate(OUTPUT_ROW_COUNT, defaultTolerance()));
+    }
+
+    @Test(enabled = false)
+    void testLineitemQ6()
+    {
+        statisticsAssertion.check(
+                "SELECT * FROM lineitem" +
+                        " WHERE l_shipdate >= DATE '1994-01-01'" +
+                        " AND l_shipdate < DATE '1994-01-01' + INTERVAL '1' YEAR" +
+                        " AND l_discount BETWEEN .06 - 0.01 AND .06 + 0.01" +
+                        " AND l_quantity < 24",
+                checks -> checks.estimate(OUTPUT_ROW_COUNT, defaultTolerance()));
+    }
+
+    @Test
+    void testLineitemQ7()
+    {
+        statisticsAssertion.check(
+                "SELECT * FROM lineitem WHERE l_shipdate BETWEEN DATE '1995-01-01' AND DATE '1996-12-31'",
+                checks -> checks.estimate(OUTPUT_ROW_COUNT, defaultTolerance()));
+    }
+
+    @Test
+    void testNationQ7()
+    {
+        statisticsAssertion.check(
+                "SELECT n_name FROM nation WHERE n_name = 'FRANCE' OR n_name = 'GERMANY'",
+                checks -> checks.estimate(OUTPUT_ROW_COUNT, defaultTolerance()));
+    }
+
+    @Test(enabled = false)
+    void testPartQ9()
+    {
+        statisticsAssertion.check(
+                "SELECT p_name FROM part WHERE p_name LIKE '%green%'",
+                checks -> checks.estimate(OUTPUT_ROW_COUNT, defaultTolerance()));
+    }
+
+    @Test(enabled = false)
+    void testLineitemQ10()
+    {
+        statisticsAssertion.check(
+                "SELECT * FROM lineitem WHERE l_returnflag = 'R'",
+                checks -> checks.estimate(OUTPUT_ROW_COUNT, defaultTolerance()));
+    }
+
+    @Test
+    void testOrdersQ13()
+    {
+        statisticsAssertion.check(
+                "SELECT * FROM orders WHERE o_comment NOT LIKE '%special%requests%'",
+                checks -> checks.estimate(OUTPUT_ROW_COUNT, defaultTolerance()));
+    }
+
+    @Test
+    void testPartQ16()
+    {
+        statisticsAssertion.check(
+                "SELECT * FROM part" +
+                        " WHERE p_brand <> 'Brand#45'" +
+                        " AND p_type NOT LIKE 'MEDIUM POLISHED%'" +
+                        " AND p_size IN (49, 14, 23, 45, 19, 3, 36, 9)",
+                checks -> checks.estimate(OUTPUT_ROW_COUNT, defaultTolerance()));
+    }
+
+    @Test(enabled = false)
+    void testSupplierQ16()
+    {
+        statisticsAssertion.check(
+                "SELECT * FROM supplier WHERE s_comment LIKE '%Customer%Complaints%'",
+                checks -> checks.estimate(OUTPUT_ROW_COUNT, defaultTolerance()));
+    }
+
+    @Test(enabled = false)
```

Comment:
That's intentional. These tests fail in 179t. They were written in the hope that as we support more constructs for stats, they would pass and can be enabled. See https://github.com/Teradata/presto/pull/667#discussion_r131742734.